index.d.ts

```diff
@@ -0,0 +1,284 @@
+/// <reference types="node" />
+
+import * as fs from "fs";
+
+export function compareSync(path1: string, path2: string, options?: Partial<Options>): Partial<Statistics>;
+export function compare(path1: string, path2: string, options?: Partial<Options>): Promise<Partial<Statistics>>;
```

Comment:
You might want to avoid returning a `Partial<>` if possible. If someone has `strictNullChecks` in their `tsconfig` then TypeScript will want them to do a `null`/`undefined` check for every property on the statistics object that they access, as it could be null.

If some stuff won't be on the `Statistics` interface all the time you can [mark it optional](https://www.typescriptlang.org/docs/handbook/interfaces.html#optional-properties).